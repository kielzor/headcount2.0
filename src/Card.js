import React from 'react';
import PropTypes from 'prop-types';


const Card = ({data, evaluateCompareCard, comparedInClass, compArrFull}) => {
  const stats = Object.entries(data.stats).map((entry, i) => {
    if (entry[1] > .5) {
      return  <div className='higher' key={`${data.location}${i}`}>
        <p className='year'>{entry[0]}</p>
        <p className='val'>{entry[1]}</p>
      </div>;
    } else {
      return  <div className='lower' key={`${data.location}${i}`}>
        <p className='year'>{entry[0]}</p>
        <p className='val'>{entry[1]}</p>
        <p className='hide'>----</p>
      </div>;
    }
  });

  const evaluateClass = () => {
    let futureClass = 'card';

    if (!compArrFull) futureClass += ' card-hover';
    if (comparedInClass) futureClass += ' selected card-hover';
    if (compArrFull && !comparedInClass) futureClass += ' no-click';

    return futureClass;
  };

  return (
    <span className={ evaluateClass() }
      onClick={() => { evaluateCompareCard(data.location) }}>
      <h5 className='location'>{data.location}</h5>
      <div className='stats'>{stats}</div>
    </span>
  );
};

Card.propTypes = {
  data: PropTypes.object,
  evaluateComparedCard: PropTypes.func,
  comparedInClass: PropTypes.bool,
  compArrFull: PropTypes.bool
};

export default Card;