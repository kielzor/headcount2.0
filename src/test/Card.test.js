import React from 'react'
import { shallow, mount } from 'enzyme'
import Card from '../Card'

describe('card', () => {
  let wrapper
  let evaluateCompareMock
  let mockData

  beforeEach(() => {
    mockData = {location: 'COLORADO', stats: {}}
    evaluateCompareMock = jest.fn()
    wrapper = shallow(<Card 
                    evaluateCompareCard={evaluateCompareMock} 
                    data={mockData}/>)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // it('adds the correct class when evaluateClass is invoked', () => {
  //   wrapper = shallow(<Card 
  //     evaluateCompareCard={evaluateCompareMock} 
  //     data={mockData}
  //     comparedInClass={true}/>)

  //     wrapper.instance().evaluateClass()

  //     expect(wrapper.futureClass).toEqual('card')
  // })

  // it('invokes evaluateCompareCard', () => {
  //   wrapper = mount(<Card 
  //     evaluateCompareCard={evaluateCompareMock} 
  //     data={mockData}/>)

  //   wrapper.find('span').simulate('click')

  //   expect(evaluateCompareMock).toHaveBeenCalled()
  // })
})