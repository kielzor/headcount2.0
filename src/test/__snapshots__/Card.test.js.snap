// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "location": "COLORADO",
            "stats": Object {},
          }
    }
    evaluateCompareCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="location"
>
          COLORADO
</h5>,
        <div
          className="stats"
>
          
</div>,
      ],
      "className": "card card-hover",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
          "className": "location",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "stats",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="location"
>
            COLORADO
</h5>,
          <div
            className="stats"
>
            
</div>,
        ],
        "className": "card card-hover",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
            "className": "location",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "stats",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
